src/main/java/net/neoforged/neoforge/registries/GameData.java

```diff
@@ -66,7 +66,7 @@ public static void unfreezeData() {
     public static void freezeData() {
         LOGGER.debug(REGISTRIES, "Freezing registries");
         BuiltInRegistries.REGISTRY.stream().filter(r -> r instanceof MappedRegistry).forEach(r -> {
-            // HolderSet.Named may be used for registry objects, vanilla binds these tags so freeze won't throw for unbound tags
+            // HolderSet.Named may be used for registry objects, vanilla binds these tags so freeze doesn't throw for unbound tags
             ((MappedRegistry<?>) r).bindAllTagsToEmpty();
```

Comment:
With the patch in `MappedRegistry` above, do we still need the `bindAllTagsToEmpty` call here?  Are they trying to achieve the same purpose?

Reply:
Different purposes. Binding all tags to empty here allows tags to be used and they'll be properly rebound on world load. The clear call is used for when a registry error occurs, where it just removes all unbound holder sets, effectively reducing them to the same result as you would get from `HolderSet.emptyNamed`.

Reply:
Makes sense, are we certain there are no other problematic call sites where we call `freezeRegistry` that might need special treatment?

Reply:
I ran a search after this last error and didn't see any other problematic call sites, but I'll double check when I get back

Reply:
Everything looks good, I'll resolve this and re-request review